src/main/java/de/tum/in/www1/artemis/service/connectors/BitbucketBambooUpdateService.java

```diff
@@ -0,0 +1,112 @@
+
+    @Value("${artemis.version-control.url}")
+    private URL BITBUCKET_SERVER;
+
+    private final Logger log = LoggerFactory.getLogger(BitbucketBambooUpdateService.class);
+
+    private final BambooClient bambooClient;
+
+    private final BambooBuildPlanUpdateProvider bambooBuildPlanUpdateProvider;
+
+    public BitbucketBambooUpdateService(BambooClient bambooClient, BambooBuildPlanUpdateProvider bambooBuildPlanUpdateProvider) {
+        this.bambooClient = bambooClient;
+        this.bambooBuildPlanUpdateProvider = bambooBuildPlanUpdateProvider;
+    }
+
+    @Override
+    public void updatePlanRepository(String bambooProject, String planKey, String bambooRepositoryName, String bitbucketProject, String bitbucketRepository,
+            Optional<List<String>> triggeredBy) {
+        try {
+            log.debug("Update plan repository for build plan " + planKey);
+            com.appfire.bamboo.cli.objects.RemoteRepository bambooRemoteRepository = bambooClient.getRepositoryHelper().getRemoteRepository(bambooRepositoryName, planKey, false);
+            // Workaround for old exercises which used a different repositoryName
+            if (bambooRemoteRepository == null) {
+                bambooRemoteRepository = bambooClient.getRepositoryHelper().getRemoteRepository("Assignment", planKey, false);
+                if (bambooRemoteRepository == null) {
+                    throw new BambooException("Something went wrong while updating the template repository of the build plan " + planKey
+                            + " to the student repository : Could not find assignment nor Assignment repository");
+                }
+            }
+
+            bambooBuildPlanUpdateProvider.updateRepository(bambooRemoteRepository, bitbucketRepository, bitbucketProject, planKey);
+
+            // Overwrite triggers if needed, incl workaround for different repo names
+            if (triggeredBy.isPresent() && bambooRemoteRepository.getName().equals("Assignment")) {
+                triggeredBy = Optional.of(triggeredBy.get().stream().map(trigger -> trigger.replace(Constants.ASSIGNMENT_REPO_NAME, "Assignment")).collect(Collectors.toList()));
+            }
+            triggeredBy.ifPresent(repoTriggers -> overwriteTriggers(planKey, bambooClient, repoTriggers));
+
+            log.info("Update plan repository for build plan " + planKey + " was successful");
+        }
+        catch (CliClient.ClientException | CliClient.RemoteRestException e) {
+            throw new BambooException("Something went wrong while updating the template repository of the build plan " + planKey + " to the student repository : " + e.getMessage(),
+                    e);
+        }
+    }
+
+    private void overwriteTriggers(final String planKey, final BambooClient bambooClient, final List<String> triggeredBy) {
+        try {
+            final var triggersString = bambooClient.getTriggerHelper().getTriggerList(planKey, null, null, 99, Pattern.compile(".*"));
```

Comment:
Can the [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) be avoided?

Reply:
See my other comment